### PR TITLE
Full Swift 4.2 compatibility

### PR DESCRIPTION
**Pull request by HeEAaD**
# Resolved issues
* Fix build error of LiveQueryDemo in Xcode 10.2 beta1
```
SWIFT_VERSION '3.0' is unsupported, supported versions are: 4.0, 4.2, 5.0. (in target 'LiveQueryDemo')
```
* Fix build errors of ParseLiveQuery when imported with Cocoapods in Xcode 10.2 beta1
```
SWIFT_VERSION ‘3.2’ is unsupported, supported versions are: 4.0, 4.2, 5.0. (in target ‘Bolts-Swift’)
SWIFT_VERSION ‘3.2’ is unsupported, supported versions are: 4.0, 4.2, 5.0. (in target ‘ParseLiveQuery’)
```

# Changes
* convert all targets to Swift 4.2
* set `swift_version` in the Podspec to 4.2

# Successful test cases
* Build of schemes "ParseLiveQuery-iOS" and "LiveQueryDemo" in Xcode 10.1 and Xcode 10.2.beta
* Build of  "ParseLiveQuery" in Xcode 10.1 and Xcode 10.2.beta when integrated with Cocapods and Podfile entry
```
pod 'ParseLiveQuery', :git => 'https://github.com/HeEAaD/ParseLiveQuery-iOS-OSX', :branch => 'swift-4.2'
```

### CODE REVIEW

**Comment by flovilmart** [comment]
Thanks for the PR. It seems that Travis is miscondugured. Do you believe we could address it before merging?

**Comment by HeEAaD** [comment]
> Thanks for the PR. It seems that Travis is miscondugured. Do you believe we could address it before merging?

The coocapods lane is empty for pull requests therefore Travis don't know what to do and it fails.
I added an explicit script step before the cocoapods distribution.

**Comment by flovilmart** [comment]
Thus is perfect! Thank!

**Review by flovilmart — APPROVED** [review]

**Comment by maxkattner** [comment]
@flovilmart Thanks a lot for your quick review! Do you think you could publish that under a new version to cocoapods so that we can bump our dependencies in our podspecs? That would help us very much :)

**Comment by flovilmart** [comment]
CAN you make a Pr with the new version number m? Then we’ll be able to publish with a tag. Added you as well as a maintainer

**Comment by maxkattner** [comment]
@flovilmart Oh yes, sure! I opened #178 just now. Please let me know if something is missing.

**Pull request merged**
